### PR TITLE
BUG fix bugs in tests that surfaced with sklearn 1.2 and autopep8 2.0.1

### PR DESCRIPTION
**Pull request by MiroDudik**
Signed-off-by: Miro Dudik <mdudik@gmail.com>

## Description

Three sklearn 1.2 related issues are being fixed:

* A typo in our test scripts: mismatched feature names between fit and predict. This was previously FutureWarning in sklearn, but with sklearn 1.2 it is now an error.
* Default behavior of log loss in sklearn 1.2 changes, affecting the evaluation of log loss in derived metrics. The default only kicks in when prediction=0 and truth=1 or vice versa, so the tests are adjusted to avoid this case (and give the log loss value equal to the previous default behavior).
* Warnings issued by fetch_openml when `parser` keyword argument is not present.

Additionally:

* autopep8 version is being kept lower to keep our linting scripts working

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: test/unit/metrics/test_generated_metrics.py [hunk 1]
```diff
@@ -7,8 +7,12 @@
 
 # ======================================================
 
+apx0 = 1e-15
+apx1 = 1 - apx0
```
Comment (riedgar-ms) [inline test/unit/metrics/test_generated_metrics.py:11]: Should this be handled through `pytest.approx()` ?

Comment (MiroDudik) [reply test/unit/metrics/test_generated_metrics.py:11]: This is not used in assertions, this is used in the construction of an alternate y_pred. The goal is to avoid taking sklearn.log_loss([0], [1]) (which would mathematically result in +infinity, but sklearn has a default behavior that just changed between two versions).

Comment (riedgar-ms) [reply test/unit/metrics/test_generated_metrics.py:11]: Could you add a comment about that?

**Review by romanlutz — APPROVED** [review]

File: test/unit/metrics/test_generated_metrics.py (hunk 1 above)
Comment (MiroDudik) [reply test/unit/metrics/test_generated_metrics.py:18]: done!

File: examples/plot_plotting_metrics_with_error.py [hunk 2]
```diff
@@ -26,14 +26,14 @@
 from fairlearn.metrics import MetricFrame
 
 # %%
-# We download the data set using :meth:`fetch_openml` function in :class:`sklearn.datasets`.
+# We download the data set using :func:`fetch_adult` function in :mod:`fairlearn.datasets`.
```
Comment (romanlutz) [inline examples/plot_plotting_metrics_with_error.py:29]: using *the*

**Review by romanlutz — APPROVED** [review]

File: examples/plot_plotting_metrics_with_error.py (hunk 2 above)
Comment (MiroDudik) [reply examples/plot_plotting_metrics_with_error.py]: done.

**Comment by MiroDudik** [comment]
The current version of fairlearn.datasets actually requires sklearn >= 1.2. Are we comfortable with that requirement?

For example, errors in other-ml job are due to the fact that their conda environments use sklearn<1.2.

I'm happy to tweak fairlearn.datasets so they work with both sklearn>=1.2 as well as with the previous versions of sklearn. Do we want that?

File: fairlearn/datasets/_fetch_acs_income.py [hunk 3]
```diff
@@ -181,6 +181,7 @@ def fetch_acs_income(
         cache=cache,
         as_frame=True,
         return_X_y=False,
+        parser="liac-arff",
```
Comment (adrinjalali) [inline fairlearn/datasets/_fetch_acs_income.py:184]: `liac-arff` is much slower than pandas though, do we really need it?
